scripts/generate_verify_psa_macros.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+
+"""Generate verify_psa_macros.c"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+
+OUTPUT_TEMPLATE = '''\
+/* Automatically generated by generate_verify_psa_macros.py. DO NOT EDIT. */
+
+#include <stdio.h>
+
+#include "psa/crypto.h"
+
+{header}
+
+#define hash_alg   PSA_ALG_SHA_256
+#define mac_alg    PSA_ALG_CMAC
+#define ka_alg     PSA_ALG_ECDH
+#define kdf_alg    PSA_ALG_HKDF(PSA_ALG_SHA_256)
+#define mac_length 4
+#define group      0x2A
+#define curve      0x2A
+#define type       PSA_KEY_TYPE_RSA_PUBLIC_KEY
+
+int main() {{
+    int ret = 0;
+
+    {tests}
+
+    return ret;
+}}
+'''
+
+HEADER_TEMPLATE = '''\
+#ifdef {name}
+#define SPEC_{name}{parameters} {expansion}
+#else
+#define {name}{parameters} {expansion}
+#endif
+'''
+
+TEST_TEMPLATE = '''\
+#ifdef SPEC_{name}
+    if ({name}{parameters} != SPEC_{name}{parameters}) {{
+        fprintf(stderr, "ERROR: %s doesn't match value in spec\\n", "{name}{parameters}");
+        ret = 1;
+    }}
+#else
+    printf("WARN:  %s is not defined\\n", "{name}{parameters}");
+#endif
+'''
+
+TEST_DEFINED_TEMPLATE = '''\
+#ifndef {name}
+    printf("WARN:  %s is not defined\\n", "{name}{parameters}");
+#endif
+'''
+
+class MacroCollector():
+    """Collect PSA crypto macro definitions from C header(-like) files.
+
+    1. Call `read_file` on the input header file(s).
+    2. Call `write_file` to write ``psa_constant_names_generated.c``.
+
+    Based on the MacroCollector in generate_psa_constants.py,
+    but *just* different enough not to warrant inheriting anything from it.
+
+    Should probably create a shared base class for the two of them at some point.
```

Comment:
Yes. Let's use this script to validate that we get the values right, but not merge it until we've cleaned it up and got rid of code duplication.